python/cuml/common/array.py

```diff
@@ -317,3 +327,12 @@ def ones(cls, shape, dtype='float32', order='F'):
             Whether to create a F-major or C-major array.
         """
         return CumlArray.full(value=1, shape=shape, dtype=dtype, order=order)
+
+
+def _check_low_level_type(data):
+    if isinstance(data, (DeviceBuffer, int, bytearray, bytes, memoryview,
+                         Buffer)):
```

Comment:
I wonder if this can be simplified a bit. Perhaps just checking if the object supports `__array_interface__` or `__cuda_array_interface__`? Admittedly you may still want to special case `DeviceBuffer` and `Buffer`.